src/core/qgsmaplayerserverproperties.cpp

```diff
@@ -0,0 +1,238 @@
+{
+  return url == other.url &&
+         type == other.type &&
+         format == other.format;
+}
+
+void QgsServerMetadataUrlProperties::clone( QgsServerMetadataUrlProperties *properties ) const
+{
+  properties->setMetadataUrls( metadataUrls() );
+}
+
+void QgsServerMetadataUrlProperties::reset()
+{
+  mMetadataUrls.clear();
+}
+
+void QgsServerMetadataUrlProperties::readXml( const QDomNode &layer_node )
+{
+  QDomElement element = layer_node.namedItem( QStringLiteral( "metadataUrls" ) ).toElement();
+  mMetadataUrls.clear();
+  const QDomNodeList el = element.elementsByTagName( QStringLiteral( "metadataUrl" ) );
+  for ( int i = 0; i < el.size(); i++ )
+  {
+    element = el.at( i ).toElement();
+    QgsMapLayerServerProperties::MetadataUrl oneUrl;
+    oneUrl.type = element.attribute( QStringLiteral( "type" ) );
+    oneUrl.format = element.attribute( QStringLiteral( "format" ) );
+    oneUrl.url = element.text();
+    addMetadataUrl( oneUrl );
+  }
+}
+
+void QgsServerMetadataUrlProperties::writeXml( QDomNode &layer_node, QDomDocument &document ) const
+{
+  QDomElement urls = document.createElement( QStringLiteral( "metadataUrls" ) );
+  for ( const QgsMapLayerServerProperties::MetadataUrl &url : mMetadataUrls )
+  {
+    QDomElement urlElement = document.createElement( QStringLiteral( "metadataUrl" ) );
+    urlElement.setAttribute( QStringLiteral( "type" ), url.type );
+    urlElement.setAttribute( QStringLiteral( "format" ), url.format );
+    urlElement.appendChild( document.createTextNode( url.url ) );
+    urls.appendChild( urlElement );
+  }
+  layer_node.appendChild( urls );
+}
+
+// QgsServerWmsDimensionProperties
+
+void QgsServerWmsDimensionProperties::clone( QgsServerWmsDimensionProperties *properties ) const
+{
+  properties->setWmsDimensions( wmsDimensions() );
+}
+
+void QgsServerWmsDimensionProperties::reset()
+{
+  mWmsDimensions.clear();
+}
+
+void QgsServerWmsDimensionProperties::setWmsDimensions( const QList<QgsServerWmsDimensionProperties::WmsDimensionInfo> &dimensions )
+{
+  mWmsDimensions = dimensions;
+}
+
+QMap<int, QString> QgsServerWmsDimensionProperties::wmsDimensionDefaultDisplayLabels()
+{
+  QMap<int, QString> labels;
+  labels[QgsServerWmsDimensionProperties::WmsDimensionInfo::AllValues] = QObject::tr( "All values" );
+  labels[QgsServerWmsDimensionProperties::WmsDimensionInfo::MinValue] = QObject::tr( "Min value" );
+  labels[QgsServerWmsDimensionProperties::WmsDimensionInfo::MaxValue] = QObject::tr( "Max value" );
+  labels[QgsServerWmsDimensionProperties::WmsDimensionInfo::ReferenceValue] = QObject::tr( "Reference value" );
+  return labels;
+}
+
+bool QgsServerWmsDimensionProperties::addWmsDimension( const QgsServerWmsDimensionProperties::WmsDimensionInfo &wmsDimInfo )
+{
+  for ( const QgsServerWmsDimensionProperties::WmsDimensionInfo &dim : mWmsDimensions )
+  {
+    if ( dim.name == wmsDimInfo.name )
```

Comment:
can we use `indexOf` in this case?

Reply:
> can we use `indexOf` in this case?

Hi @pblottiere 

The search  predicat is based on the attribut `name` attribut, `indexOf`  is  based on contained object equality semantic (there is no way to pass a functor or lambda as search predicat) .

Reply:
Yes indeed.